### PR TITLE
[NEW] Render video preview on video upload.

### LINKED ISSUE #766: [IMPROVEMENT] No video preview picture is rendered
- Your Rocket.Chat app version: 1.1.0
- Your Rocket.Chat server version: 0.61.0
- Device (or Simulator) you're running with: Motorola Moto G

**When uploading a video file, no preview picture is created**
The video upload works fine (.3gp file), but you do not see a preview picture and an in-app video player. Rather just a link i shown. The download of the file works as expected.

### PR DESCRIPTION
**Pull request by abdulwd**
@RocketChat/android

Closes #766 

When a video is uploaded, a preview of the video is shown.
image

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt
Comment (luciofm) [inline app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt]: Please create a `fun` with a meaningful name returning the Observable.

Also, you need to unsubscribe on view recycling, or you can end up having the wrong preview when scrolling the list...

(This doesn't happens right now because we have a bug on the adapter, that will be fixed on a PR I'm sending today)

**Review by luciofm — CHANGES_REQUESTED** [review]

File: app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt
Comment (abdulwd) [reply app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt]: Sure! I was unable to work as there was some problem with my laptop.

**Comment by abdulwd** [comment]
@luciofm Recently, I pulled the latest code, so there are no merge conflicts, from **develop-2.x** branch of _Rocket.Chat.Android_ and from **develop** branch of _Rocket.Chat.Kotlin.SDK_ and have performed the build of SDK and have executed the script in **Rocket.Chat.Android/apps/** but the build of the app is failing with the following error:
```
`RocketChat/Rocket.Chat.Android/app/src/main/java/chat/rocket/android/chatrooms/presentation/ChatRoomsPresenter.kt
Error:(209, 22) Unresolved reference: lastModified
Error:(239, 51) Unresolved reference: lastModified
Error:Execution failed for task ':app:compileDebugKotlin'
> Compilation error. See log for more details`.
```
I have tried common troubleshooting steps and also cloning the repositories again and again performing the step-up for both the repositories again but still, the error persists.

**Comment by rafaelks** [comment]
@abdulwd Could you please fix the conflicts?

**Comment by abdulwd** [comment]
@rafaelks Yes! I would like to fix this issue completely but every time I pull from the SDK code and from Rocket.Chat.Android and build the code I get compile-time errors due to missing fields from SDK which are required by Rocket.Chat.Android. I have tried common troubleshooting steps but it didn't work. :disappointed:

**Comment by rafaelks** [comment]
@abdulwd The reason the SDK is not working for you is because we're working on this branch now: https://github.com/RocketChat/Rocket.Chat.Android/pull/796. Once this branch is merged, I guess it'll work for you as it was working before. 👍 

@luciofm Let's merge this release branch today?

**Comment by abdulwd** [comment]
@rafaelks Thanks. So should I rebase my branch on #796 or should I wait for #796 to get merged?

**Comment by rafaelks** [comment]
@abdulwd Everything is now merged... can you give it a try again please?

**Comment by abdulwd** [comment]
@rafaelks Sure! :smiley: 
I have my mid-semester exams until 9th March. I will try to fix this ASAP.

**Comment by abdulwd** [comment]
@rafaelks @luciofm I have made the changes.
I'm using the same approach as before but now the tests are failing. Previously, all tests were passing. Can you please help me resolve this.

**Comment by rafaelks** [comment]
@abdulwd That's probably only the CI... let me run it again.

**Comment by abdulwd** [comment]
@rafaelks @luciofm I have just resolved the conflicts and force pushed the changes. Now all the checks pass. Don't know why this happened. Please review. :smile:

**Comment by leonardoaramaki** [comment]
@luciofm Looks good to me but wouldn't it be better if we use coroutines here instead of rx?

**Comment by rafaelks** [comment]
@abdulwd Thanks for the PR, but I'm not being able to see the preview of the video here... do I need to do something special?

screenshot_20180329-232835

**Comment by abdulwd** [comment]
Sorry for the late reply.
@leonardoaramaki Yes we can use coroutines but then we have to use a try-catch expression.
@rafaelks I'm just fetching the first frame so in your case the first frame might be black. You can try another video.
A better approach is to have an API which could provide the thumbnail of the video. As presently, there is no caching and I am fetching the full resolution frame so the loading of preview takes time. If we have an API to provide the thumbnail image we can use Fresco to load the thumbnail and it will be cached.

**Comment by leonardoaramaki** [comment]
Thanks, but we're closing this one as we're redesigning the player soon. :+1:

**Comment by wajahatshaw** [comment]
Hi! @leonardoaramaki Is this feature added yet?

**Pull request closed without merging**
